test/controllers/authentications/microsoft_graph_oauth_controller_test.rb

```diff
@@ -0,0 +1,182 @@
+require "test_helper"
+
+class Authentications::MicrosoftGraphOauthControllerTest < ActionDispatch::IntegrationTest
+  setup do
+    OmniAuth.config.test_mode = true
+  end
+
+  test "cancelling a microsoft oauth flow redirects to the settings page" do
+    login_as users(:keith)
+    get "/auth/failure"
+    assert_redirected_to edit_settings_person_path
+  end
```

Comment:
We should definitely remove the first of these two tests (logged in yet hitting /auth/failure). That is related to the same `if Current.user` comment I made within the controller. It's a flow which is unique to google.